File: day11/java/src/HourGlasses.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;

public class HourGlasses {

    public static int hourGlassSum(List<List<Integer>> arr, int x, int y){
        int suma = arr.get(y).get(x) + arr.get(y).get(x + 1) + arr.get(y).get(x + 2);
        suma += arr.get(y + 1).get(x + 1);
        suma += arr.get(y + 2).get(x) + arr.get(y + 2).get(x + 1) + arr.get(y + 2).get(x + 2);

        return suma;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        List<List<Integer>> arr = new ArrayList<>();

        IntStream.range(0, 6).forEach(i -> {
            try {
                arr.add(
                        Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                                .map(Integer::parseInt)
                                .collect(toList())
                );
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        int size = arr.size();
        int resultado = -10000;

        for (int y = 0; y < size - 2; y++){
            for(int x = 0; x < size - 2; x++){
                resultado = Math.max(resultado, HourGlasses.hourGlassSum(arr, x, y));
            }
        }

        System.out.println(resultado);

        bufferedReader.close();
    }


}

File: day11/python/hourglasses.py
#!/bin/python3

def hourglass_sum(x, y):
    suma = arr[x][y] + arr[x][y + 1] + arr[x][y + 2]
    suma += arr[x + 1][y + 1]
    suma += arr[x + 2][y] + arr[x + 2][y + 1] + arr[x + 2][y + 2]
    return suma


if __name__ == '__main__':
    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    size = len(arr)
    suma = -100000
    for x in range(0, (size - 2)):
        for y in range(0, (size - 2)):
            k = hourglass_sum(x, y)
            if k > suma:
                suma = k
    print(suma)
